### PR TITLE
Add support for running tests with the earliest possible simulator, and use it for introspection tests.

### PR DESCRIPTION
**Pull request by rolfbjarne**
* Add support for running tests with the earliest possible simulator
  (currently iOS 8.1, tvOS 9.1 and watchOS 2.0).
* Make the introspection tests run with the earliest possible simulator.
* Fix several binding issues (mostly missing availability attributes), and a
  couple of issues in the introspection tests themselves.

Reference: https://github.com/xamarin/xamarin-macios/issues/3668.

### CODE REVIEW

File: src/HealthKit/EnumsCompat.cs [hunk 1]
```diff
@@ -15,6 +15,8 @@ public unsafe static partial class HKQuantityTypeIdentifierKey  {
 			}
 		}
 
+		[Introduced (PlatformName.iOS, 9,3, PlatformArchitecture.All)]
+		[Introduced (PlatformName.WatchOS, 2,2, PlatformArchitecture.All)]
```
Comment (spouliot) [inline src/HealthKit/EnumsCompat.cs]: it's already on the `getter` ?
looks like the test code is wrong (and should be fixed)
but we should not duplicate it on both the property and the getter (too much metadata if the linker is not enabled)

Comment (rolfbjarne) [reply src/HealthKit/EnumsCompat.cs]: The generator puts the attributes on both the getter and the property:

```csharp
[BindingImpl (BindingImplOptions.GeneratedCode | BindingImplOptions.Optimizable)]
[Introduced (PlatformName.iOS, 9,0, PlatformArchitecture.All)]
[Introduced (PlatformName.MacOSX, 10,11, PlatformArchitecture.All)]
public static NSString ColorPrimaries_DCI_P3 {
	[Introduced (PlatformName.iOS, 9,0, PlatformArchitecture.All)]
	[Introduced (PlatformName.MacOSX, 10,11, PlatformArchitecture.All)]
	get {
		return CVImageBufferColorPrimaries.DciP3.GetConstant () as NSString;
	}
}
```

I agree that it should be redundant to put the attributes on both, but I think that's best fixed in a different PR.

**Comment by rolfbjarne** [comment]
> I wonder if those conflicts with devices

I ran introspection on my iOS 8.2 device and the 64-bit version worked (the 32-bit version failed to build due to the 32-bit size issue).

**Comment by rolfbjarne** [comment]
build

File: src/HealthKit/EnumsCompat.cs (hunk 1 above)
Comment (spouliot) [reply src/HealthKit/EnumsCompat.cs]: ok, there must be a reason (what the IDE display) for the duplication. Better find it (again) before fixing it :)

**Comment by rolfbjarne** [comment]
The binary mlaunch must be updated: https://github.com/xamarin/macios-binaries/pull/18.

**Comment by rolfbjarne** [comment]
Test failure is unrelated:

* mmptest: https://github.com/xamarin/maccore/issues/1597

**Pull request merged**